apis/placement/v1alpha1/disruptionbudget_types.go

```diff
@@ -0,0 +1,150 @@
+	//   value of the NumberOfClusters fields in the placement policy).
+	// The end result will be rounded up to the nearest integer if applicable.
+	//
+	// One may use a value of 0 for this field; in this case, no voluntary disruption would be
+	// allowed.
+	//
+	// This field is mutually exclusive with the MinAvailable field in the spec; exactly one
+	// of them can be set at a time.
+	//
+	// Defaults to 25%.
+	// +kubebuilder:default="25%"
+	// +kubebuilder:validation:XIntOrString
+	// +kubebuilder:validation:Pattern="^((100|[0-9]{1,2})%|[0-9]+)$"
+	// +optional
+	MaxUnavailable *intstr.IntOrString `json:"maxUnavailable,omitempty"`
+
+	// MinAvailable is the minimum number of placements that must be available at any time
+	// despite voluntary disruptions. For example, a setting of 10 would imply that
+	// a voluntary disruption (e.g., an eviction) can only happen if there are at least 11
+	// placements from the linked ClusterResourcePlacement object are applied and available.
+	//
+	// This can be either an absolute value (e.g., 1) or a percentage (e.g., 10%).
+	//
+	// If a percentage is specified, Fleet will calculate the corresponding absolutate values
+	// as follows:
+	// * if the linked ClusterResourcePlacement object is of the PickFixed placement type,
+	//   the percentage is against the number of clusters specified in the placement (i.e., the
+	//   length of ClusterNames field in the placement policy);
+	// * if the linked ClusterResourcePlacement object is of the PickAll placement type,
+	//   the percentage is against the total number of clusters being selected by the scheduler
+	//   at the time of the evaluation of the disruption budget;
+	// * if the linked ClusterResourcePlacement object is of the PickN placement type,
+	//   the percentage is against the number of clusters specified in the placement (i.e., the
+	//   value of the NumberOfClusters fields in the placement policy).
+	// The end result will be rounded up to the nearest integer if applicable.
+	//
+	// One may use a value of 0 for this field; in this case, voluntary disruption would be
+	// allowed at any time.
+	//
+	// This field is mutually exclusive with the MaxUnavailable field in the spec; exactly one
+	// of them can be set at a time.
+	//
+	// +kubebuilder:validation:XIntOrString
+	// +kubebuilder:validation:Pattern="^((100|[0-9]{1,2})%|[0-9]+)$"
+	// +optional
+	MinAvailable *intstr.IntOrString `json:"minAvailable,omitempty"`
+}
+
+// PlacementDisruptionBudgetStatus is the observed state of the
+// PlacementDisruptionBudget.
+type PlacementDisruptionBudgetStatus struct {
+	// Conditions is the list of currently observed conditions for the
+	// PlacementDisruptionBudget object.
+	//
+	// Available condition types include:
+	// * Applied: whether the disruption budget has been successfully applied to the linked
+	//   ClusterResourcePlacement object.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
```

Comment:
Do we need to provide the current status like?

total: the total number of target clusters of the placement.
currentAvailable: the number of currently available clusters.
desiredAvailable: the number of desired available clusters.
disruptionsAllowed: the number of clusters that are currently allowed to be disrupted.

Reply:
After discussing with @michaelawyu offline, the PDB is just used for information purposes. 

When the eviction controller reconciles an eviction, it needs to list all PDBs associated with the CRP re-calculates voluntary disruptions allowed by PDBs and take a decision on the eviction.

Even if we update the PDB's status with the information suggested above there is a high chance the info. is stale.

The actual decision is always taken inside the eviction controller

Reply:
Yeah, just to re-cap a few things that Arvind and I discussed:

* The eviction controller will evaluate DBs at the time of execution;
* It's definitely helpful to have status that reports info. such as max allowed disruptions, just like K8s does with PDBs; those are for informational purposes only -> considering the timeline, this does not necessarily have to be done right away

Reply:
If we have to calculate those numbers anyway, why not exposing them in the API?

It will be a puzzle for the user to understand why an eviction is rejected by PDB. Or will the eviction status contain such information?

Reply:
If the eviction status will contain these numbers and which PDB blocked the eviction, it is ok to not report these numbers here.

Reply:
Understood. 👌 The eviction object status will include the reason why an eviction couldn't go through (e.g., `failed to execute as DB abc does not allow the op: max allowed disruptions: 2, current disruptions: 3`).

The concern we had earlier was mostly timeline-based: due to freshness constraints, the eviction controller must calculate the numbers at the time of execution, yet any status we report on the DB objects will be refreshed only periodically (the eviction controller couldn't rely on that information). The latter is not on the critical path, so just to be on the safer side for delivery it might be easier for us to focus on the main flow first. These pieces of information are obviously useful to users, of course ✨

Reply:
SGTM.